validators/pool.ak

```diff
@@ -293,6 +311,8 @@ validator pool(
 
         // And make sure each of these fields is unchanged
         and {
+          datum.condition == actual_condition,
+          datum.condition_datum == actual_condition_datum,
```

Comment:
btw, we should allow the `condition_datum` to change, in case the condition wants to update it's state; the condition can be in charge of making sure it's unchanged, if that makes sense for the specific condition in question